qtwidgetsextra/additional/QShape.h

```diff
@@ -0,0 +1,81 @@
+#ifndef QSHAPE_H
+#define QSHAPE_H
+
+#include <QFrame>
+#include <QBrush>
+#include <QPen>
+#include <QPainter>
+
+class QShape : public QFrame
+{
+    Q_OBJECT
+    class QShapePrivate *d;
+    friend class QShapePrivate;
+    Q_ENUMS(Shape)
+
+    Q_PROPERTY(QPainter::RenderHints renderHints READ renderHints WRITE setRenderHints)
+    Q_PROPERTY(QMargins margins READ margins WRITE setMargins)
+    Q_PROPERTY(QBrush brush READ brush WRITE setBrush)
+    Q_PROPERTY(QPen pen READ pen WRITE setPen)
+    Q_PROPERTY(QShape::Shape shape READ shape WRITE setShape)
+    Q_PROPERTY(QPoint radius READ radius WRITE setRadius)
+
+    enum ShapeFlag {
+        CircleFlag = 0x1,
+        DiamondFlag = 0x2,
```

Comment:
why hex values and it's better to have it in the right order. especially if this has to grow. Two enum values can have the same associated values